platform/test.rs

```diff
@@ -174,30 +173,77 @@ fn big_data_with_sender_transfer() {
     thread.join().unwrap();
 }
 
-#[test]
-// This test only applies to platforms that need fragmentation.
+macro_rules! create_with_n_fds {
+    ($name:ident, $n:expr, $size:expr) => (
+        #[test]
+        fn $name() {
```

Comment:
Redundant wrappers? What do you mean?

I don't see how wrapper functions are redundant. There's a tiny amount of boilerplate, but the clarity is worth it IMO. I always have a hard time reading macros.